gradle-module-plugin/src/main/kotlin/io/ia/sdk/gradle/modl/extension/RequiredModuleDependency.kt

```diff
@@ -0,0 +1,21 @@
+package io.ia.sdk.gradle.modl.extension
+
+import java.io.Serializable
+
+class RequiredModuleDependency(
```

Comment:
Realized after seeing all the code together that `RequiredModuleDependency` and `requiredModuleDependencies` are confusing, since `required` can be--and is by default--`false`.

A straightforward change would be just renaming to something like `ModuleDependencySpecification` and `moduleDependencySpecifications` or similar, implying it's a more structured flavor of `moduleDependencies` and its `MapProperty<String,String>` type.

I'm going to see if a trick is possible to smooth this deprecation further. Will take me a bit but might pay off.

Reply:
Before I look for that trick, my branch is now synced with yours, plus this [commit at the head](https://github.com/brianeray/ignition-module-tools/commit/8e9ce856c493aa2e2b37eda60514d5fecfb4f6b7) with the renaming above.

`gradlew check` is completely clean on that refactor.